### PR TITLE
NodeJS: Add `columns()` method to get type info from prepared statement

### PR DESCRIPTION
**Pull request by Maxxen**
This PR adds a `.columns()` method on the prepared statement object to retrieve the bound column names and their type info from a query. In essence, this returns the query "schema". 
I've opted to include the type "id" as a string, along with the full sql type string. Nested types contain nested type information, and for types with extra parameters like enums and decimals, that information is provided as additional properties in the returned object, as well. 

Example excerpts from tools/nodejs/test/columns.test.ts:
```js
{ name: 'bool', type: { id: 'BOOLEAN',  sql_type: 'BOOLEAN' } },
```
```js
{
  name: 'small_enum',
  type: {
    id: 'ENUM',
    sql_type: 'small_enum',
    name: 'small_enum',
    values: [
      "DUCK_DUCK_ENUM",
      "GOOSE"
    ]
  }
},
```
```js
{
  name: 'struct',
  type: {
    id: 'STRUCT',
    sql_type: 'STRUCT(a INTEGER, b VARCHAR)',
    children: [
      { 
        name: 'a',
        type: {
          id: 'INTEGER',
          sql_type: 'INTEGER'
        }
      },
      {
        name: 'b',
        type: {
          id: 'VARCHAR',
          sql_type: 'VARCHAR'
        }
      }
    ]
  }
},
```

### CODE REVIEW

File: tools/nodejs/lib/duckdb.d.ts [hunk 1]
```diff
@@ -171,6 +171,45 @@ export class Database {
   ): Promise<void>;
 }
 
+type TypeInfo = {
+  id: string,
+  sql_type: string,
+  alias?: string,
+} | {
```
Comment (Mause) [inline tools/nodejs/lib/duckdb.d.ts:178]: These should probably be made to be standalone exported types/interfaces that are then used in TypeInfo

File: tools/nodejs/lib/duckdb.js [hunk 2]
```diff
@@ -689,6 +689,14 @@ Statement.prototype.stream;
  */
 Statement.prototype.sql;
 
+/**
+ * @method
+ * @param {...*} params
```
Comment (Mause) [inline tools/nodejs/lib/duckdb.js:694]: These tags aren't accurate? I'm not sure we need to define the full return shape in jsdoc, but at least should give the rough shape

**Review by Mause — CHANGES_REQUESTED** [review]

**Comment by Mause** [comment]
Love it! Very similar idea to the description field in the Python DB API standard, but much more flexible (I'm a bit jealous)

Most of my comments are about typing peculiarities in typescript, feel free to ask if you're not sure what I'm on about

**Comment by Maxxen** [comment]
Thanks! Yeah typescript makes it pretty nice to describe these sort of recursive/discriminated typings, I'll try to address the feedback asap. Also since this uses `test_all_types()` it will probably break #7701 if this makes it in before, but Ill file a PR to your fork and fix it in that case!

File: tools/nodejs/lib/duckdb.d.ts [hunk 3]
```diff
@@ -185,6 +224,8 @@ export class Statement {
   finalize(callback?: Callback<void>): void;
 
   run(...args: [...any, Callback<void>] | any[]): Statement;
+
+  columns(): {name: string, type: TypeInfo}[];
```
[the remark this answers is not in the document]
Comment (Maxxen) [reply tools/nodejs/lib/duckdb.d.ts]: Yes and no, semantically it's not the same, columns return the column name, while the type info children use the nested type member names. I've added aliases for both tough.

File: tools/nodejs/lib/duckdb.js (hunk 2 above)
Comment (Maxxen) [reply tools/nodejs/lib/duckdb.js]: Fixed!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**